modules/data_models.py

```diff
@@ -0,0 +1,363 @@
+"""
+This module contains classes for easy data management
+"""
+from constants import HUMIDITY_UNIT, TEMPERATURE_UNIT, WEATHER_FILES_DIR
+from modules.utils import (
+    get_date,
+    get_highest_temperature,
+    get_lowest_temperature,
+    get_max_humidity,
+    get_mean_humidity,
+    get_month_name,
+    parse_line,
+    read_data,
+)
+
+
+class DayData:
+    """
+    This class holds data of a particular day
+    """
+
+    def __init__(self, line):
+        """
+        Initializes the line member and calls populate to initialize other members
+        Args:
+            line(list): list of strings containing all data from line read of weather file
+        """
+        self.line = line
+        self.populate()
+
+    def populate(self):
+        """
+        Sets the values of highest_temperature, lowest_temperature, max_humidity,
+        mean_humidity and date by using the parsed_line
+        Returns:
+            None
+        """
+        self.highest_temperature = get_highest_temperature(self.line)
+        self.lowest_temperature = get_lowest_temperature(self.line)
+        self.max_humidity = get_max_humidity(self.line)
+        self.mean_humidity = get_mean_humidity(self.line)
+        self.date = get_date(self.line)
+
+    def get_max_temperature(self):
+        """
+        Returns the highest_temperature, can be None
+        Returns:
+            (int or None):  highest_temperature in degree celsius
+                            Or
+                            None
+        """
+
+        return self.highest_temperature
+
+    def get_min_temperature(self):
+        """
+        Returns the lowest_temperature, can be None
+        Returns:
+            (int or None):  lowest_temperature in degree celsius
+                            Or
+                            None
+        """
+
+        return self.lowest_temperature
+
+    def get_max_humidity(self):
+        """
+        Returns the max_humidity, can be None
+        Returns:
+            (int or None):  max_humidity in percentage
+                            Or
+                            None
+        """
+
+        return self.max_humidity
+
+    def get_mean_humidity(self):
+        """
+        Returns the mean_humidity, can be None
+        Returns:
+            (int or None):  mean_humidity in percentage
+                            Or
+                            None
+        """
+
+        return self.mean_humidity
+
+    def get_date(self):
```

Comment:
Can you set the date in `datetime object` format which will make the date-related operations more flexible usage?

Reply:
Done